### PR TITLE
Allow to consume multiple reports from a tar archive

### PR DESCRIPTION
**Pull request by dLobatog**
This is meant to support upload of multiple reports at once in a tar -  https://github.com/RedHatInsights/insights-core/pull/2415

### CODE REVIEW

File: app/consumers/compliance_reports_consumer.rb [hunk 1]
```diff
@@ -42,14 +42,14 @@ def download_file
...
+
+    logger.info "Received message, enqueueing: #{@msg_value}"
+    @report_contents.each do |report|
       job = ParseReportJob.perform_async(
-        ActiveSupport::Gzip.compress(@report_contents), @msg_value
+        ActiveSupport::Gzip.compress(report), @msg_value
       )
       logger.info "Message enqueued: #{message_id} as #{job}"
-    else
-      logger.error "Error parsing report: #{message_id}"
```
Comment (dLobatog) [inline app/consumers/compliance_reports_consumer.rb]: Notice this is the *exact* same error as the one that's logged in `validation_message`  if there's a failure. It makes no sense to log the same thing twice.

File: app/services/reports_tar_reader.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+require 'rubygems/package'
+require 'zlib'
+
+# Finds the files in a insights tar and returns them as an array
+class ReportsTarReader
+  def initialize(file)
+    @file = file
+  end
+
+  def reports
```
Comment (dLobatog) [inline app/services/reports_tar_reader.rb]: I'll add a sample tarfile here created with `insights-core` to test this one

File: app/services/reports_tar_reader.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    tar_extract = Gem::Package::TarReader.new(Zlib::GzipReader.open(@file))
+    tar_extract.rewind # The extract has to be rewinded after every iteration
+    files = []
+    tar_extract.each do |entry|
+      files << entry.read
+    end
+    tar_extract.close
```
Comment (akofink) [inline app/services/reports_tar_reader.rb]: This should probably be in an `ensure` block at the end of this method.

Comment (akofink) [inline app/services/reports_tar_reader.rb]: This seems like a great opportunity to use `yield` and wrap all this TarReader stuff in its own method which can contain the `ensure` to close the archive.

File: app/consumers/compliance_reports_consumer.rb [hunk 4]
```diff
@@ -60,7 +60,9 @@ def validate
   end
 
   def validation_message
-    XccdfReportParser.new(@report_contents, @msg_value)
+    @report_contents.each do |report|
+      XccdfReportParser.new(report, @msg_value)
+    end
```
Comment (akofink) [inline app/consumers/compliance_reports_consumer.rb]: Should we allow partial processing of an archive if one of the reports fails? The StandardError rescue below will halt parsing for all remaining reports in the archive if that error is raised for any one report.

**Pull request merged**